diff --git a/mesonbuild/envconfig.py b/mesonbuild/envconfig.py
index 4055b2176..4a7c746c2 100644
--- a/mesonbuild/envconfig.py
+++ b/mesonbuild/envconfig.py
@@ -363,6 +363,12 @@ class MachineInfo(HoldableObject):
         """
         return self.system == 'aix'
 
+    def is_zos(self) -> bool:
+        """
+        Machine is z/OS?
+        """
+        return self.system == 'zos'
+
     def is_irix(self) -> bool:
         """Machine is IRIX?"""
         return self.system.startswith('irix')
