diff --git a/mesonbuild/linkers/detect.py b/mesonbuild/linkers/detect.py
index cf6803a19..619c8d071 100644
--- a/mesonbuild/linkers/detect.py
+++ b/mesonbuild/linkers/detect.py
@@ -220,6 +220,16 @@ def guess_nix_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         linker = linkers.AIXDynamicLinker(
             compiler, env, for_machine, comp_class.LINKER_PREFIX, override,
             version=search_version(e))
+    elif 'IEW5033' in e:
+        if isinstance(comp_class.LINKER_PREFIX, str):
+            _, o, e = Popen_safe(compiler + [comp_class.LINKER_PREFIX + '-V'] + extra_args)
+        else:
+            _, o, e = Popen_safe(compiler + comp_class.LINKER_PREFIX + ['-V'] + extra_args)
+        version = re.search(r'z/OS (V\d+ R\d+) BINDER', o)
+        version = 'unknown version' if version is None else version.group(1)
+        linker = linkers.ZOSDynamicLinker(
+            compiler, env, for_machine, comp_class.LINKER_PREFIX, override,
+            version=version)
     elif o.startswith('zig ld'):
         linker = linkers.ZigCCDynamicLinker(
             compiler, env, for_machine, comp_class.LINKER_PREFIX, override, version=v)
